src/structure.js

```diff
@@ -493,35 +518,11 @@ function hasIn(cursor, path) {
   return cursor.getIn(path, NOT_SET) !== NOT_SET;
 }
 
-function listMatchingOrCreateEmptyList (listeners, path) {
-  var pathArray = [];
-  return (path || []).reduce(function(acc, pathPart) {
-    pathArray = pathArray.concat(pathPart);
-    var listenerData = listMatchingOrCreateEmptyListOnNamespace(acc.listeners, pathArray);
-
-    return {
-      listeners: listenerData.listeners,
-      matched: acc.matched.concat(listenerData.matched)
-    };
-  }, {
-    listeners: listeners,
-    matched: []
-  });
-}
-
-function listMatchingOrCreateEmptyListOnNamespace (listeners, path) {
-  path = (path || []).concat(LISTENER_SENTINEL);
-  var matched = listeners.getIn(path);
-
-  if (!matched) {
-    matched = [];
-    listeners = listeners.setIn(path, matched);
```

Comment:
I think this is pretty key, it represents the whole theory of the perf gains we accomplished with #55 that idea being having a tree structure to store callbacks and traversing that rather than everything to get the functions to call. Which it appears has been wiped out in these changes.